File: MyDailyBeat/MyDailyBeat/EVCFirstTimePreferencesViewController.m
//
//  EVCFirstTimePreferencesViewController.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 7/12/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "EVCFirstTimePreferencesViewController.h"
#import "VervePreferences.h"
#import <RESideMenu.h>
#import "EVCViewController.h"

@interface EVCFirstTimePreferencesViewController ()

@end

@implementation EVCFirstTimePreferencesViewController

@synthesize api;

- (void)viewDidLoad {
    [super viewDidLoad];
    self.formController = [[FXFormController alloc] init];
    self.formController.tableView = self.tableView;
    self.formController.delegate = self;
    self.formController.form = [[VervePreferences alloc] init];
    api = [RestAPI getInstance];
    
    [self retrievePrefs];
}

- (void) retrievePrefs {
    dispatch_queue_t queue = dispatch_queue_create("dispatch_queue_t_dialog", NULL);
    dispatch_async(queue, ^{
        VervePreferences *prefs = [[VervePreferences alloc] init];
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.view makeToastActivity];
        });
        prefs.userPreferences = [api getUserPreferencesForUser:[api getCurrentUser]];
        prefs.matchingPreferences = [api getMatchingPreferencesForUser:[api getCurrentUser]];
        prefs.hobbiesPreferences = [api getHobbiesPreferencesForUserWithScreenName:[api getCurrentUser].screenName];
        self.formController.form = prefs;
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.view hideToastActivity];
            [self.tableView reloadData];
        });
    });
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void) selectEthnicity:(UITableViewCell<FXFormFieldCell> *)cell {
    VervePreferences *prefs = cell.field.form;
    self.formController.form = prefs;
    NSLog(@"Hello World");
    [self.tableView reloadData];
}

- (void) selectBeliefs:(UITableViewCell<FXFormFieldCell> *)cell {
    VervePreferences *prefs = cell.field.form;
    self.formController.form = prefs;
    [self.tableView reloadData];
}

- (void)submit:(UITableViewCell<FXFormFieldCell> *)cell {
    VervePreferences *prefs = cell.field.form;
    dispatch_queue_t queue = dispatch_queue_create("dispatch_queue_t_dialog", NULL);
    dispatch_async(queue, ^{
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.view makeToastActivity];
        });
        
        BOOL success = [api saveUserPreferences:prefs.userPreferences andMatchingPreferences:prefs.matchingPreferences forUser:[api getCurrentUser]];
        BOOL success2 = [api saveHobbiesPreferences:prefs.hobbiesPreferences forUserWithScreenName:[api getCurrentUser].screenName];
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.view hideToastActivity];
            if (success && success2) {
                [[NSUserDefaults standardUserDefaults] setBool:NO forKey:@"FirstTimeLogin"];
                [self.navigationController popToRootViewControllerAnimated:YES];
            } else
                NSLog(@"Failed");
            
        });
    });
}

@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCFirstTimePreferencesViewController.swift

//
//  EVCFirstTimePreferencesViewController.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 7/12/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import UIKit
import Toast_Swift
import API
import FXForms
class EVCFirstTimePreferencesViewController: UIViewController, FXFormControllerDelegate {
    @IBOutlet var tableView: UITableView!
    var api: RestAPI!
    var formController: FXFormController!


    override func viewDidLoad() {
        super.viewDidLoad()
        self.formController = FXFormController()
        self.formController.tableView = self.tableView
        self.formController.delegate = self
        self.formController.form = VervePreferences()
        api = RestAPI.getInstance()
        self.retrievePrefs()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(false, animated: false)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
    }

    func retrievePrefs() {
        
        DispatchQueue.global().async(execute: {() -> Void in
            var prefs = VervePreferences()
            DispatchQueue.main.async(execute: {() -> Void in
                self.view.makeToastActivity(ToastPosition.center)
            })
            prefs.userPreferences = self.api.getUserPreferences()
            prefs.matchingPreferences = self.api.getMatchingPreferences()
            prefs.hobbiesPreferences = self.api.getHobbiesPreferencesForUser()
            self.formController.form = prefs
            DispatchQueue.main.async(execute: {() -> Void in
                self.view.hideToastActivity()
                self.tableView.reloadData()
            })
        })
    }

    func submit(_ cell: FXFormBaseCell) {
        let prefs: VervePreferences? = cell.field.form as! VervePreferences?
        
        DispatchQueue.global().async(execute: {() -> Void in
            DispatchQueue.main.async(execute: {() -> Void in
                self.view.makeToastActivity(ToastPosition.center)
            })
            let success: Bool = self.api.save((prefs?.userPreferences)!, andMatchingPreferences: (prefs?.matchingPreferences)!)
            let success2 = self.api.save((prefs?.hobbiesPreferences)!)
            DispatchQueue.main.async(execute: {() -> Void in
                self.view.hideToastActivity()
                if success && success2 {
                    self.dismiss(animated: true, completion: nil)
                }
                else {
                    print("Failed")
                }
            })
        })
    }
}
